src/Neo/SmartContract/ApplicationEngine.cs

```diff
@@ -180,7 +180,16 @@ public virtual UInt160 CallingScriptHash
         protected unsafe ApplicationEngine(
             TriggerType trigger, IVerifiable container, DataCache snapshotCache, Block persistingBlock,
             ProtocolSettings settings, long gas, IDiagnostic diagnostic, JumpTable jumpTable = null)
-            : base(jumpTable ?? DefaultJumpTable)
+            : base(
+               // this is IsHardforkEnabled
+               settings is not null
+                   ? (persistingBlock is null
+                       ? settings.Hardforks.ContainsKey(Hardfork.HF_Echidna)
```

Comment:
Only with `IsHardforkEnabled` is not enough?

Reply:
Ping @Jim8y

Reply:
casue it  can not use IsHardforkEnabled here, its an instance methods, while what we need is a class level method that can be used directly beofre the constructor.

Reply:
Fixed